src/_pytest/recwarn.py

```diff
@@ -83,7 +84,7 @@ def deprecated_call(
 
 @overload
 def warns(
-    expected_warning: Optional[Union[Type[Warning], Tuple[Type[Warning], ...]]],
+    expected_warning: Union[Type[Warning], Tuple[Type[Warning], ...]] = ...,
```

Comment:
```suggestion
    expected_warning: Union[Type[Warning], Tuple[Type[Warning], ...]],
```

Reply:
I had to add this again, as mypy was complaining that there is no overload of `warns` that allows for no arguments to be passed. The `...` signifies the default argument that is passed in [here](https://github.com/olgarithms/pytest/blob/warns-no-arg-catches-any/src/_pytest/recwarn.py#L105).